backend.py

```diff
@@ -265,7 +283,20 @@ def create_result(self, std_out, std_err, status="intermediate", files=False):
             for name in filenames:
                 # ToDO: if filesize > 1mb -> generate s3-url
                 file_path = os.path.join(self.tmp_dir, "files", name)
-                mimetype = mime.from_file(file_path)
+                mime = mimetypes.guess_type(file_path)
+                mimetype = "" 
+                if mime[0] == "text/plain":
+                    with open(file_path, 'r') as fr:
+                        lines = fr.readlines()
+                    is_uri = all([re.match(self.regex, line.strip()) for line in lines])
+                    if is_uri:
+                        mimetype = "text/uri-list"
+                    else: 
+                        mimetype = "text/octet-stream"
+                elif mime[0] is not None:
+                    mimetype = mime[0]
+                else:
+                    mimetype = "text/octet-stream"
```

Comment:
Is there a reason why you choose "text" octet-stream instead of "application" octet-stream ? It can be also a binary file, can't it?